igvc_navigation/src/mapper/main.cpp

```diff
@@ -13,221 +15,281 @@
 #include <opencv2/core/eigen.hpp>
 #include <opencv2/opencv.hpp>
 #include "tf/transform_datatypes.h"
+#include <limits.h>
+#include <pcl_conversions/pcl_conversions.h>
 
-igvc_msgs::map msgBoi;  // >> message to be sent
 cv_bridge::CvImage img_bridge;
-sensor_msgs::Image imageBoi;  // >> image in the message
 
 ros::Publisher map_pub;
 ros::Publisher debug_pub;
-ros::Subscriber odom_sub;
 ros::Publisher debug_pcl_pub;
-cv::Mat *published_map;  // matrix will be publishing
+//ros::Subscriber odom_sub;
+std::unique_ptr<cv::Mat> published_map;  // matrix will be publishing
 std::map<std::string, tf::StampedTransform> transforms;
-tf::TransformListener *tf_listener;
-std::string topics;
-Eigen::Map<Eigen::Matrix<unsigned char, Eigen::Dynamic, Eigen::Dynamic>> *eigenRep;
+std::unique_ptr<tf::TransformListener> tf_listener;
 
 double resolution;
 double orientation;
 double _roll;
 double _pitch;
+double transform_max_wait_time;
```

Comment:
I don't remember we have decided on a naming scheme for global vars. But it seems like that some vars are already named starting with '_' for member variable. maybe we should address that change as well?

Reply:
I think a recent commit to devel has changed independent roll, pitch, cur_x, cur_y into a `RobotState` object, for consistency try to incorporate that here.